kolibri/plugins/user/assets/src/views/SignInPage.vue

```diff
@@ -282,11 +282,12 @@
       ...mapState(['facilityId']), // backend's default facility on load
       ...mapState('signIn', ['hasMultipleFacilities', 'usersForSelectedFacilities']),
       ...mapState({
+        passwordMissing: state => state.core.loginError === LoginErrors.PASSWORD_MISSING,
         invalidCredentials: state => state.core.loginError === LoginErrors.INVALID_CREDENTIALS,
         busy: state => state.core.signInBusy,
       }),
       simpleSignIn() {
-        return this.facilityConfig.learner_can_login_with_no_password;
+        return this.facilityConfig.learner_can_login_with_no_password && !this.passwordMissing;
```

Comment:
To clarify here - this is being toggled off in the case that a non-learner has tried to login without the password, and then the backend has reported that the password is needed?

Reply:
I have moved it to the computed property to decide if the password must be shown, to be more semantic (if that can be said in English)

Reply:
Yep - perfectly semantic use of semantic for me :)